model/Service/TimeoutCommand.php

```diff
@@ -35,23 +35,14 @@ final class TimeoutCommand implements
     use NavigationContextAwareTrait;
     use ToolsStateAwareTrait;
 
-    /** @var QtiRunnerServiceContext */
-    private $serviceContext;
-
-    /** @var bool */
-    private $hasStartTimer;
-
-    /** @var bool */
-    private $lateSubmissionAllowed;
+    public const FEATURE_FLAG_TIMEOUT_PERMANENT_LATE_SUBMISSION = 'FEATURE_FLAG_TIMEOUT_PERMANENT_LATE_SUBMISSION';
 
     public function __construct(
-        QtiRunnerServiceContext $serviceContext,
-        bool $hasStartTimer,
-        bool $lateSubmissionAllowed
+        private readonly QtiRunnerServiceContext $serviceContext,
+        private readonly bool $hasStartTimer,
+        private readonly bool $lateSubmissionAllowed,
+        private readonly bool $permanentLateSubmission = false,
```

Comment:
Are we sure, that all customers are using PHP 8.1+ to support readonly syntax?
If yes - we are ok 🙂